docs/usage/shoot_pod_autoscaling_best_practices.md

```diff
@@ -0,0 +1,327 @@
+- If you have defined absolute limits (unrelated to the requests), configure VPA to only scale the requests or else it will proportionally scale the limits as well, which can easily become useless (way beyond the threshold of unhealthy behavior) or absurd (larger than node capacity):
+  ```
+  spec:
+    resourcePolicy:
+      containerPolicies:
+      - controlledValues: RequestsOnly
+        ...
+  ```
+  If you have defined relative limits (related to the requests), the default policy to scale the limits proportionally with the requests is fine, but the gap between requests and limits must be zero for QoS `Guaranteed` and should best be small for QoS `Burstable` to avoid useless or absurd limits either, e.g. prefer limits being 5 to at most 20% larger than requests as opposed to being 100% larger or more.
+- As a rule of thumb, set `minAllowed` to the highest observed VPA recommendation (usually during the initialization phase or during any periodical activity) for an otherwise practically idle container, so that you avoid needless trashing (e.g. resource usage calms down over time and recommendations drop consecutively until eviction, which will then lead again to initialization or later periodical activity and higher recommendations and new evictions).<br/>⚠️ You may want to provide higher `minAllowed` values, if you observe that up-scaling takes too long for CPU or memory for a too large percentile of your workload. This will get you out of the danger zone of too few resources for too many pods at the expense of providing too many resources for a few pods. Memory may react faster than CPU, because CPU throttling is not visible and memory gets aided by OOM bump-up incidents, but still, if you observe that up-scaling takes too long, you may want to increase `minAllowed` accordingly.
+- As a rule of thumb, set `maxAllowed` to your theoretical maximum load, flanked with alerts to detect erroneous run-away usage or the actual nearing of your practical maximum load, so that you can intervene. However, VPA can easily recommend requests larger than what is allocatable on a node, so you must either ensure large enough nodes (Gardener can scale up from zero, in case you like to define a low-priority worker pool with more resources for very large pods) and/or cap VPA's target recommendations using `maxAllowed` at the node allocatable remainder (after daemonset pods) of the largest eligible machine type (may result in under-provisioning resources for a pod). Use your monitoring and check maximum pod usage to decide about the maximum machine type.
+
+## Recommendations in a Box
+
+| Container | When to use | Value |
```

Comment:
Hmm... actually I wanted to say container, also because I wrote container as the unit of work above in the text. Resource is somewhat ambiguous is it not?

Reply:
In the k8s world `container` usually refers to the OCI container, therefore to avoid ambiguity I proposed `resource` or `resource requirement` as this is the term used in the k8s API and docs:
- https://github.com/kubernetes/kubernetes/blob/0fa4b9e14cba6342cfaaea4db793e36e4b8ae20d/staging/src/k8s.io/api/core/v1/types.go#L2730-L2734
- https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/

Reply:
Yes, I know the term, but it isn't helpful in my eyes. It is part of the more relevant [container spec](https://github.com/kubernetes/kubernetes/blob/0fa4b9e14cba6342cfaaea4db793e36e4b8ae20d/staging/src/k8s.io/api/core/v1/types.go#L2660) (encompassing entity) and I find the term "container" therefore more useful in this table as a short column header than the in-the-middle-of-the-spec term `resource requirements`. The docs don't mention the term `resource requirement` at all (until the end/API ref link) and start with "Resource Management for Pods and **Containers**" and then delve right into the next title "**Requests** and **limits**".

I aim to reach also uninitiated people and they know what a container is and maybe they know that it has requests and limits, but they do not really care about this term from the internal structures and the general term `resource` is far to ambiguous/not helpful. The more familiar term is container and I want them to think about pods and their containers.

So, sorry, I will not implement this suggestion, because I do not feel this will improve the document.